### PR TITLE
r/sageamaker_model_package_group - new resource

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSSagemakerModelPackageGroup_'
--- PASS: TestAccAWSSagemakerModelPackageGroup_disappears (26.29s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_basic (36.78s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_description (36.89s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_tags (89.19s)
...
```

### CODE REVIEW

File: aws/resource_aws_sagemaker_model_package_group.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		input.ModelPackageGroupDescription = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		input.Tags = keyvaluetags.New(v.(map[string]interface{})).IgnoreAws().SagemakerTags()
+	}
+
+	_, err := conn.CreateModelPackageGroup(input)
+	if err != nil {
+		return fmt.Errorf("error creating Sagemaker Model Package Group %s: %w", name, err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model_package_group.go:71]:
```suggestion
		return fmt.Errorf("error creating SageMaker Model Package Group %s: %w", name, err)
```

File: aws/resource_aws_sagemaker_model_package_group.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	mpg, err := finder.ModelPackageGroupByName(conn, d.Id())
+	if err != nil {
+		if isAWSErr(err, "ValidationException", "does not exist") {
+			d.SetId("")
+			log.Printf("[WARN] Unable to find Sagemaker Model Package Group (%s); removing from state", d.Id())
+			return nil
+		}
+		return fmt.Errorf("error reading Sagemaker Model Package Group (%s): %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model_package_group.go:94]:
```suggestion
		return fmt.Errorf("error reading SageMaker Model Package Group (%s): %w", d.Id(), err)
```

File: aws/resource_aws_sagemaker_model_package_group.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+	arn := aws.StringValue(mpg.ModelPackageGroupArn)
+	d.Set("model_package_group_name", mpg.ModelPackageGroupName)
+	d.Set("arn", arn)
+	d.Set("model_package_group_description", mpg.ModelPackageGroupDescription)
+
+	tags, err := keyvaluetags.SagemakerListTags(conn, arn)
+
+	if err != nil {
+		return fmt.Errorf("error listing tags for Sagemaker Model Package Group (%s): %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model_package_group.go:106]:
```suggestion
		return fmt.Errorf("error listing tags for SageMaker Model Package Group (%s): %w", d.Id(), err)
```

File: aws/resource_aws_sagemaker_model_package_group.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+func resourceAwsSagemakerModelPackageGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+
+		if err := keyvaluetags.SagemakerUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating Sagemaker Model Package Group (%s) tags: %s", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model_package_group.go:123]:
```suggestion
			return fmt.Errorf("error updating SageMaker Model Package Group (%s) tags: %s", d.Id(), err)
```

File: aws/resource_aws_sagemaker_model_package_group.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+	input := &sagemaker.DeleteModelPackageGroupInput{
+		ModelPackageGroupName: aws.String(d.Id()),
+	}
+
+	if _, err := conn.DeleteModelPackageGroup(input); err != nil {
+		if isAWSErr(err, "ValidationException", "does not exist") {
+			return nil
+		}
+		return fmt.Errorf("error deleting Sagemaker Model Package Group (%s): %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model_package_group.go:141]:
```suggestion
		return fmt.Errorf("error deleting SageMaker Model Package Group (%s): %w", d.Id(), err)
```

File: aws/resource_aws_sagemaker_model_package_group.go [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	if _, err := waiter.ModelPackageGroupDeleted(conn, d.Id()); err != nil {
+		if isAWSErr(err, "ValidationException", "does not exist") {
+			return nil
+		}
+		return fmt.Errorf("error waiting for Sagemaker Model Package Group (%s) to delete: %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_model_package_group.go:148]:
```suggestion
		return fmt.Errorf("error waiting for SageMaker Model Package Group (%s) to delete: %w", d.Id(), err)
```

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerModelPackageGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerModelPackageGroup_ -timeout 120m
=== RUN   TestAccAWSSagemakerModelPackageGroup_basic
=== PAUSE TestAccAWSSagemakerModelPackageGroup_basic
=== RUN   TestAccAWSSagemakerModelPackageGroup_description
=== PAUSE TestAccAWSSagemakerModelPackageGroup_description
=== RUN   TestAccAWSSagemakerModelPackageGroup_tags
=== PAUSE TestAccAWSSagemakerModelPackageGroup_tags
=== RUN   TestAccAWSSagemakerModelPackageGroup_disappears
=== PAUSE TestAccAWSSagemakerModelPackageGroup_disappears
=== CONT  TestAccAWSSagemakerModelPackageGroup_basic
=== CONT  TestAccAWSSagemakerModelPackageGroup_disappears
=== CONT  TestAccAWSSagemakerModelPackageGroup_tags
=== CONT  TestAccAWSSagemakerModelPackageGroup_description
--- PASS: TestAccAWSSagemakerModelPackageGroup_description (16.91s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_disappears (17.04s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_basic (21.84s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_tags (42.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	42.671s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerModelPackageGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerModelPackageGroup_ -timeout 120m
=== RUN   TestAccAWSSagemakerModelPackageGroup_basic
=== PAUSE TestAccAWSSagemakerModelPackageGroup_basic
=== RUN   TestAccAWSSagemakerModelPackageGroup_description
=== PAUSE TestAccAWSSagemakerModelPackageGroup_description
=== RUN   TestAccAWSSagemakerModelPackageGroup_tags
=== PAUSE TestAccAWSSagemakerModelPackageGroup_tags
=== RUN   TestAccAWSSagemakerModelPackageGroup_disappears
=== PAUSE TestAccAWSSagemakerModelPackageGroup_disappears
=== CONT  TestAccAWSSagemakerModelPackageGroup_basic
=== CONT  TestAccAWSSagemakerModelPackageGroup_disappears
=== CONT  TestAccAWSSagemakerModelPackageGroup_tags
=== CONT  TestAccAWSSagemakerModelPackageGroup_description
--- PASS: TestAccAWSSagemakerModelPackageGroup_disappears (11.47s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_basic (15.51s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_description (15.71s)
--- PASS: TestAccAWSSagemakerModelPackageGroup_tags (34.09s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	35.855s
```

Failing Acceptance Tests in GovCloud (us-gov-west-1)

These tests will need to be skipped, a follow up issue will be created to track this work.

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerModelPackageGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerModelPackageGroup_ -timeout 120m
=== RUN   TestAccAWSSagemakerModelPackageGroup_basic
=== PAUSE TestAccAWSSagemakerModelPackageGroup_basic
=== RUN   TestAccAWSSagemakerModelPackageGroup_description
=== PAUSE TestAccAWSSagemakerModelPackageGroup_description
=== RUN   TestAccAWSSagemakerModelPackageGroup_tags
=== PAUSE TestAccAWSSagemakerModelPackageGroup_tags
=== RUN   TestAccAWSSagemakerModelPackageGroup_disappears
=== PAUSE TestAccAWSSagemakerModelPackageGroup_disappears
=== CONT  TestAccAWSSagemakerModelPackageGroup_basic
=== CONT  TestAccAWSSagemakerModelPackageGroup_disappears
=== CONT  TestAccAWSSagemakerModelPackageGroup_description
=== CONT  TestAccAWSSagemakerModelPackageGroup_tags
=== CONT  TestAccAWSSagemakerModelPackageGroup_disappears
    resource_aws_sagemaker_model_package_group_test.go:163: Step 1/1 error: Error running apply: 2021/02/10 08:18:46 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.12.0

        Error: error creating SageMaker Model Package Group tf-acc-test-8314827973041593631: UnknownOperationException: The requested operation is not supported in the called region.
        	status code: 400, request id: 0a6c864d-d77e-43a2-91f5-e906eefe46f8


=== CONT  TestAccAWSSagemakerModelPackageGroup_description
    resource_aws_sagemaker_model_package_group_test.go:93: Step 1/2 error: Error running apply: 2021/02/10 08:18:46 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.12.0

        Error: error creating SageMaker Model Package Group tf-acc-test-3740237711744291030: UnknownOperationException: The requested operation is not supported in the called region.
        	status code: 400, request id: 3069b992-f9b9-4f85-a736-93685db778c4


=== CONT  TestAccAWSSagemakerModelPackageGroup_tags
    resource_aws_sagemaker_model_package_group_test.go:119: Step 1/4 error: Error running apply: 2021/02/10 08:18:46 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.12.0

        Error: error creating SageMaker Model Package Group tf-acc-test-4268414703224873836: UnknownOperationException: The requested operation is not supported in the called region.
        	status code: 400, request id: fe2dbb2c-14bf-4491-ae55-039686df8caa


=== CONT  TestAccAWSSagemakerModelPackageGroup_basic
    resource_aws_sagemaker_model_package_group_test.go:65: Step 1/2 error: Error running apply: 2021/02/10 08:18:46 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.12.0

        Error: error creating SageMaker Model Package Group tf-acc-test-6501003966160909864: UnknownOperationException: The requested operation is not supported in the called region.
        	status code: 400, request id: 076128f1-f182-4a92-b063-cb25303e0085


--- FAIL: TestAccAWSSagemakerModelPackageGroup_disappears (4.20s)
--- FAIL: TestAccAWSSagemakerModelPackageGroup_description (4.21s)
--- FAIL: TestAccAWSSagemakerModelPackageGroup_tags (4.22s)
--- FAIL: TestAccAWSSagemakerModelPackageGroup_basic (4.23s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	6.039s
FAIL
make: *** [testacc] Error 1
```

**Pull request merged**